### PR TITLE
Fixes #9365 : let with exportC pragma in template.

### LINKED ISSUE #9365: [OpenMP] Nim symbol interpolation support
Similar to emit pragma it would be very useful to have Nim symbol interpolation support in OpenMP annotations.

Currently the following code:

```Nim
import sequtils, macros

var x = toSeq(1..1000)

let ompsize = x.len
let ompthreshold = 800

for i in `||`(0, x.len - 1, "if(`ompthreshold` < `ompsize`)"):
  x[i] += 2

echo x[0]
```

Generates:
```C
// ... removing seq initialization
	T7_ = (x_wFAR9b1Cy8ShaaTQG2FH9c7g ? x_wFAR9b1Cy8ShaaTQG2FH9c7g->Sup.len : 0);
	ompsize_89bqUwA9bOswTIcWIVyaG9cfw = T7_;
	ompthreshold_XaMtJe9ajIFJ9btvwMO3AaEw = ((NI) 800);
	T8_ = (x_wFAR9b1Cy8ShaaTQG2FH9c7g ? x_wFAR9b1Cy8ShaaTQG2FH9c7g->Sup.len : 0);
	#pragma omp parallel for if(`ompthreshold` < `ompsize`)
for (i = ((NI) 0); i <= (NI)(T8_ - ((NI) 1)); ++i)	{
		x_wFAR9b1Cy8ShaaTQG2FH9c7g->data[i] += ((NI) 2);
	}
	nimZeroMem((void*)T10_, sizeof(tyArray_nHXaesL0DJZHyVS07ARPRA));
	T10_[0] = nimIntToStr(x_wFAR9b1Cy8ShaaTQG2FH9c7g->data[((NI) 0)]);
	echoBinSafe(T10_, 1);
```

### PR DESCRIPTION
**Pull request by deech**

### CODE REVIEW

File: compiler/semtempl.nim [hunk 1]
```diff
@@ -197,6 +197,13 @@ proc addLocalDecl(c: var TemplCtx, n: var PNode, k: TSymKind) =
     else:
       replaceIdentBySym(c.c, n, ident)
   else:
+    if (n.kind == nkPragmaExpr and sonsLen(n) >= 2 and n.sons[1].kind == nkPragma):
+      let pragmaNode = n.sons[1]
+      let numPragmas = sonsLen(pragmaNode)
+      for i in 0 .. numPragmas-1:
```
Comment (Araq) [inline compiler/semtempl.nim]: This loop should start at ``1``.

Comment (Araq) [reply compiler/semtempl.nim]:
```suggestion

for i in 1..<pragmaNode.len:

```

Comment (deech) [reply compiler/semtempl.nim]: Loop needs to start at `0` so the first pragma isn't ignored but I used your suggestion iterating through.

Comment (Araq) [reply compiler/semtempl.nim]: ah right, I misread, my bad.

**Comment by mratsim** [comment]
I think that fixes https://github.com/nim-lang/Nim/issues/9366 not https://github.com/nim-lang/Nim/issues/9365

**Comment by deech** [comment]
Yes you're right, my mistake.

**Pull request merged**